### ISSUE TITLE
Getting circuit cost from user input

### ISSUE THREAD

**Issue by aszady** [qiskit-community/Quantum-Challenge-Grader#1]
This code suggests that the grading server might accept the score calculated by the client, which not necessarily must be the real one.

https://github.com/qiskit-community/IBMQuantumChallenge2020-Grader/blob/fd231c86d71d6e59d2542085b3abd69329c66b79/qc_grader/grade.py#L226

As the server's code is not public, few experiments were made showing that indeed cost different from the actual one (higher or lower) is accepted and used later in the ranking. Such design makes the live ranking unreliable.

**Comment by HuangJunye** [comment]
@aszady Thanks for reporting this as well! We were aware of the risk of cheating by manipulating the the qc_cost. This design was used to offload the computation of cost from server to the client. Only at the last stage we came up with the current compute cost method which can compute cost a fraction of a second instead of minutes. We could have hidden the compute cost back to the server but it was too close to the challenge that we decided to keep it as it is. As mentioned in #2, we will manually check the top 10 solutions and verify the cost using Unroller to make sure the ranking is accurate.
